### PR TITLE
MAM-3626-consider-use-url-search-on-instance-to-get-quote-post-from

### PR DESCRIPTION
**Pull request by derspyy**
allows us to render every fediverse object as a quote post, rather than only mastodon stuff, also as a bonus makes quoting conform to the instance's moderation, since the user's instance fetches the post.

this is mainly a POC, currently incompatible with ivory because they just append the link under the post.

images: (can i make these smaller?)
image
image

### CODE REVIEW

File: Mammoth/Managers/StatusCache.swift [hunk 1]
```diff
@@ -125,59 +125,23 @@ class StatusCache {
...
+            Task {
+                do {
+                    let result = try await ClientService.runRequest(request: request)
+                    if let stat = result.statuses.first {
+                        await MainActor.run {
                             StatusCache.cachedStatusesLock.lock()
                             StatusCache.cachedStatuses[url] = stat
                             StatusCache.cachedStatusesLock.unlock()
                         }
+                        completion(url, stat)
```
Comment (bnolens) [inline Mammoth/Managers/StatusCache.swift:138]: Please add a `log.error("couldn't find post.")` statement or something similar

Comment (derspyy) [reply Mammoth/Managers/StatusCache.swift:138]: will do

**Comment by derspyy** [comment]
image

i've seen this case where the url card seems to take preference, but it can't say if this is particularly _wrong_ behaviour. do we want to overwrite these?

**Comment by derspyy** [comment]
image

this is also very useful to view quotes from akkoma -> mastodon. this is how they'd look originally:

image

**Review by bnolens — APPROVED** [review]
Great work @derspyy. LGTM! I made one small change to limit the QuotePost search to only 1 result.

**Comment by derspyy** [comment]
> Great work @derspyy. LGTM! I made one small change to limit the QuotePost search to only 1 result. 

cool, no problem

**Pull request merged**